pkg/storage/snapshot/source.go

```diff
@@ -286,17 +286,12 @@ func (s *vmSnapshotSource) Spec() (snapshotv1.SourceSpec, error) {
 }
 
 func (s *vmSnapshotSource) Online() (bool, error) {
-	vmRunning, err := checkVMRunning(s.vm)
```

Comment:
I get that the runstrategy now doesnt promise us the status of the vm if it is stopped/stopping/starting/started.
But the existence of VMI doesnt ensure it either since its racefull.. I am a bit concerned of unexpected behaviors in case of someone stopping/starting a VM and do vmsnapshot at the same time. WDYT?

Reply:
I don't think there is any way to avoid a race. In the case you describe, the controller may attempt to freeze the VM and fail but on subsequent reconcile, `source.Online()` will be false and freeze will be skipped.

Reply:
Im more worried on the opposite case where it will skip freeze because it was marked as offline..

Reply:
What's the problem there exactly?  If the snapshot is completed before the VMI exists it is technically an offline snapshot.

Reply:
If the vmi didnt exist right at the time where the freeze didnt occur and then was created and written to then the snapshot is taken then the snapshot might not be valid.. I know its a longshot and possibly a very (very) small window but theoretically possible

Reply:
I'm not sure how we can entirely eliminate that race without having some higher level synchronization. Creating a restore and starting a VMI are fundamentally two independent operations.  Checking VM runStrategy is not helpful and can cause problems like described in the linked issue. The goal of this PR is to eliminate the downsides of checking runStrategy

Reply:
You are right this is a race we can't entirely eliminate. So I'm not against merging this PR, but how about thinking of a way to check if there was a change in the snapshot "online/offline" vm state after "locking" the vm from changing state since as we said its a bit racefull. Something like comparing the state of the vm before "unlocking" it and see if it changed comparing to how we expected it to be, specifically if we marked it as offline but it got online somehow and if so to mark the snapshot as failed.

Reply:
Yeah maybe that is something we can handle in another PR